articles/clean-and-cleaner-code.md

```diff
@@ -0,0 +1,28 @@
+*	Focus - Good code should follow the Single responsibility principle (SRP), so that the code we write should only have one reason to change and to be small, or to say that is responsible of a single part of functionality. 
+*	Simplicity - We should always try to apply DRY (Don’t repeat yourself) rule. What this means is to reduce repetition of code, so that a single change does not produce many changes in code. In addition, we should follow the KISS principle (Keep It Simple Stupid), which force us not to complicate things unnecessary.
+*	Readability - In order to make code more readable, we can follow the YAGNI principle (You Ain’t Gonna Need It), which implies that we shouldn’t write code upfront that is not currently needed.
+*	Naming - Naming things is one of the most important things in software development. Names are everywhere in software. We name methods, classes, files, etc. It is important to give meaningful names as they need to reveal intent. Names should be expressive enough and clear that we immediately know what concept represents. This also represent a self-documenting approach.
+*	Commenting - As comments usually represent an anti-pattern we should be careful with them. If we have many comments, this usually means that the code is not self-documenting. We need to tend to focus on more clear and readable code and not to have many comments which explains code.
+*	Formatting - We always first tend to make a working software, but in the long run, we will need to have good readability. To achieve this, our code needs to have a coding style understandable by a team, because without it maintainability can be seriously affected, especially on large projects. Although these days we have powerful tools that can take care of formatting, teams should adopt formatting rules and follow them.
+
+Along with these guidelines a developer should use “code sense” when looking at code, according to Robert C. Martin [1], the creator of Clean code approach. This let people to see different options and that they can select the best variation of it, to make value-added and clean code. To do that, we must practice coding and explore different dimensions of programming languages. 
+It is also important reminder that we need to write nice and clean code, but also the code that solves the problem. Certainly, we need to deliver code with or without clean code, than not to deliver it at all. However, the importance of clean code should always be a top priority, as this will bring more benefits to a project in the future. It should become a practice to go back and clean the code before delivery.
+
+1. Clean Code: A Handbook of Agile Software Craftsmanship (Robert C. Martin), Prentice Hall, 2008.
+
+
+*By {{page.authorName}}*
+
+{% include "./_example.resources.md" %}
```

Comment:
Replace with {% include "./clean-and-cleaner-code.md.resources.md" %}
and put the reference to 
_Clean Code: A Handbook of Agile Software Craftsmanship (Robert C. Martin), Prentice Hall, 2008._
into this file

On the other hand, is this reference really required? The referencing text is "Along with these guidelines a developer should use “code sense” when looking at code, according to Robert C. Martin [1], the creator of Clean code approach" -> you could also change this text to just "Along with these guidelines a developer should use “code sense” when looking at code."